### ISSUE TITLE
Lane fails in gym via Jenkins

### ISSUE THREAD

**Issue by evilDave** [fastlane/fastlane#5092]
### Issue Description

All is working running "locally"
Running though Jenkins, it has an error:

```
INFO [2016-06-10 17:29:36.88]: ▸ ❌  Code Sign error: No codesigning identities found: No codesigning identities (i.e. certificate and private key pairs) that match the provisioning profile specified in your build settings (“match AdHoc com.hotelscombined.findhotels.beta”) were found.
INFO [2016-06-10 17:29:36.88]: ▸ ** ARCHIVE FAILED **
```

It seems like it gets the provisioning profile. I have the signing identity set to "iOSDistribution". I notice that when I run this as myself, the iPhone Distribution certificate is added to my login keychain, but running as jekins it is not added to the jenkins (user's) keychain.

If I run `match adhoc -a com.hotelscombined.findhotels.beta --readonly` it completes with `All required keys, certificates and provisioning profiles are installed 🙌` but there is no distribution certificate in the keychain...

##### Complete output when running fastlane, including the stack trace and command used

Running the same command as the jenkins used gives this

```
bash-3.2$ fastlane beta_HC --verbose
DEBUG [2016-06-10 17:29:16.69]: Checking if there are any plugins that should be loaded...
INFO [2016-06-10 17:29:16.70]: -------------------------------------------------
INFO [2016-06-10 17:29:16.70]: --- Step: Verifying required fastlane version ---
INFO [2016-06-10 17:29:16.70]: -------------------------------------------------
INFO [2016-06-10 17:29:16.70]: fastlane version valid
INFO [2016-06-10 17:29:16.71]: ------------------------------
INFO [2016-06-10 17:29:16.71]: --- Step: default_platform ---
INFO [2016-06-10 17:29:16.71]: ------------------------------
INFO [2016-06-10 17:29:16.74]: Driving the lane 'ios beta_HC' 🚀
INFO [2016-06-10 17:29:16.74]: -------------------------------------
INFO [2016-06-10 17:29:16.74]: --- Step: Switch to ios beta lane ---
INFO [2016-06-10 17:29:16.74]: -------------------------------------
INFO [2016-06-10 17:29:16.74]: Cruising over to lane 'ios beta' 🚖
INFO [2016-06-10 17:29:16.74]: -------------------
INFO [2016-06-10 17:29:16.74]: --- Step: match ---
INFO [2016-06-10 17:29:16.74]: -------------------

+-----------------------+---------------------------------------------------------+
|                             Summary for match 0.6.1                             |
+-----------------------+---------------------------------------------------------+
| type                  | adhoc                                                   |
| readonly              | true                                                    |
| git_url               | git@github.com:evilDave/HotelsCombined-Certificates.git |
| git_branch            | master                                                  |
| app_identifier        | com.hotelscombined.findhotels.beta                      |
| username              | [REDACTED]                          |
| keychain_name         | login.keychain                                          |
| team_id               | [REDACTED]                                              |
| verbose               | false                                                   |
| force                 | false                                                   |
| shallow_clone         | false                                                   |
| force_for_new_devices | false                                                   |
| skip_docs             | false                                                   |
+-----------------------+---------------------------------------------------------+

INFO [2016-06-10 17:29:17.02]: Cloning remote git repo...
INFO [2016-06-10 17:29:17.02]: $ git clone 'git@github.com:evilDave/HotelsCombined-Certificates.git' '/var/folders/sl/rb8wdyyx2cn6_fpxz_vqd_j800007c/T/d20160610-23600-1wr1rrn'
INFO [2016-06-10 17:29:17.04]: ▸ Cloning into '/var/folders/sl/rb8wdyyx2cn6_fpxz_vqd_j800007c/T/d20160610-23600-1wr1rrn'...
INFO [2016-06-10 17:29:20.72]: ▸ remote: Counting objects: 36, done.
INFO [2016-06-10 17:29:20.98]: ▸ remote: Compressing objects: 100% (26/26), done.
INFO [2016-06-10 17:29:21.34]: ▸ remote: Total 36 (delta 3), reused 31 (delta 1), pack-reused 0
INFO [2016-06-10 17:29:21.34]: ▸ Receiving objects: 100% (36/36), 57.03 KiB | 0 bytes/s, done.
INFO [2016-06-10 17:29:21.34]: ▸ Resolving deltas: 100% (3/3), done.
INFO [2016-06-10 17:29:21.66]: ▸ Checking connectivity... done.
INFO [2016-06-10 17:29:21.73]: 🔓  Decrypted '2W3WN5GQB7.cer'
INFO [2016-06-10 17:29:21.76]: 🔓  Decrypted '2W3WN5GQB7.p12'
INFO [2016-06-10 17:29:21.80]: 🔓  Decrypted 'AdHoc_com.hotelscombined.biyi.*.mobileprovision'
INFO [2016-06-10 17:29:21.83]: 🔓  Decrypted 'AdHoc_com.hotelscombined.findhotels.beta.mobileprovision'
INFO [2016-06-10 17:29:21.83]: 🔓  Successfully decrypted certificates repo
INFO [2016-06-10 17:29:21.83]: Installing certificate...
INFO [2016-06-10 17:29:21.83]: $ security list-keychains -d user
INFO [2016-06-10 17:29:21.84]: ▸     "/Users/Shared/Jenkins/jenkins.keychain"
INFO [2016-06-10 17:29:21.84]: $ security find-certificate -c 'Apple Worldwide Developer Relations Certification Authority' /Users/Shared/Jenkins/jenkins.keychain
INFO [2016-06-10 17:29:21.87]: ▸ keychain: "/Users/Shared/Jenkins/jenkins.keychain"
INFO [2016-06-10 17:29:21.87]: ▸ version: 256
INFO [2016-06-10 17:29:21.87]: ▸ class: 0x80001000
INFO [2016-06-10 17:29:21.87]: ▸ attributes:
INFO [2016-06-10 17:29:21.87]: ▸     "alis"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2016-06-10 17:29:21.87]: ▸     "cenc"<uint32>=0x00000003
INFO [2016-06-10 17:29:21.87]: ▸     "ctyp"<uint32>=0x00000001
INFO [2016-06-10 17:29:21.87]: ▸     "hpky"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2016-06-10 17:29:21.87]: ▸     "issu"<blob>=0x3062310B300906035504061302555331133011060355040A130A4150504C4520494E432E31263024060355040B131D4150504C452043455254494649434154494F4E20415554484F52495459311630140603550403130D4150504C4520524F4F54204341  "0b1\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\023\012APPLE INC.1&0$\006\003U\004\013\023\035APPLE CERTIFICATION AUTHORITY1\0260\024\006\003U\004\003\023\015APPLE ROOT CA"
INFO [2016-06-10 17:29:21.87]: ▸     "labl"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2016-06-10 17:29:21.87]: ▸     "skid"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2016-06-10 17:29:21.87]: ▸     "snbr"<blob>=0x01DEBCC4396DA010  "\001\336\274\3049m\240\020"
INFO [2016-06-10 17:29:21.87]: ▸     "subj"<blob>=0x308196310B300906035504061302555331133011060355040A0C0A4170706C6520496E632E312C302A060355040B0C234170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E733144304206035504030C3B4170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E732043657274696669636174696F6E20417574686F72697479  "0\201\2261\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\014\012Apple Inc.1,0*\006\003U\004\013\014#Apple Worldwide Developer Relations1D0B\006\003U\004\003\014;Apple Worldwide Developer Relations Certification Authority"
INFO [2016-06-10 17:29:22.04]: $ security import /var/folders/sl/rb8wdyyx2cn6_fpxz_vqd_j800007c/T/d20160610-23600-1wr1rrn/certs/distribution/2W3WN5GQB7.cer -k ~/Library/Keychains/login.keychain -T /usr/bin/codesign -T /usr/bin/security&> /dev/null
INFO [2016-06-10 17:29:22.07]: $ security import /var/folders/sl/rb8wdyyx2cn6_fpxz_vqd_j800007c/T/d20160610-23600-1wr1rrn/certs/distribution/2W3WN5GQB7.p12 -k ~/Library/Keychains/login.keychain -T /usr/bin/codesign -T /usr/bin/security&> /dev/null
INFO [2016-06-10 17:29:22.10]: Installing provisioning profile...
WARN [2016-06-10 17:29:22.20]: Setting environment variable 'sigh_com.hotelscombined.findhotels.beta_adhoc' to 'c70fdd4f-ef88-49dd-a2e0-14f42645c69b'

+----------------------+-----------------------------------------------+
|                    Installed Provisioning Profile                    |
+----------------------+-----------------------------------------------+
| App Identifier       | com.hotelscombined.findhotels.beta            |
| Type                 | adhoc                                         |
| UUID                 | [REDACTED]                                    |
| Environment Variable | sigh_com.hotelscombined.findhotels.beta_adhoc |
+----------------------+-----------------------------------------------+

INFO [2016-06-10 17:29:22.20]: All required keys, certificates and provisioning profiles are installed 🙌
INFO [2016-06-10 17:29:22.20]: 🔒  Successfully encrypted certificates repo
INFO [2016-06-10 17:29:22.20]: Setting Provisioning Profile type to 'ad-hoc'
INFO [2016-06-10 17:29:22.21]: -----------------
INFO [2016-06-10 17:29:22.21]: --- Step: gym ---
INFO [2016-06-10 17:29:22.21]: -----------------
WARN [2016-06-10 17:29:22.26]: Using legacy build system - waiting for radar to be fixed: https://openradar.appspot.com/radar?id=4952000420642816
WARN [2016-06-10 17:29:22.73]: xcrun xcodebuild -list -workspace './HCWeb.xcworkspace'

+----------------------+----------------------+
|            Summary for gym 1.6.3            |
+----------------------+----------------------+
| scheme               | HCWeb Beta           |
| use_legacy_build_api | true                 |
| export_method        | ad-hoc               |
| workspace            | ./HCWeb.xcworkspace  |
| destination          | generic/platform=iOS |
| output_name          | HCWeb                |
| clean                | false                |
| output_directory     | .                    |
| silent               | false                |
| buildlog_path        | ~/Library/Logs/gym   |
+----------------------+----------------------+

+-----------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------+
|                                                                               Generated Build Command                                                                               |
+-----------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------+
| Option                                                                      | Value                                                                                                 |
+-----------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------+
| set -o pipefail &&                                                          |                                                                                                       |
| xcodebuild                                                                  |                                                                                                       |
| -workspace                                                                  | ./HCWeb.xcworkspace                                                                                   |
| -scheme                                                                     | HCWeb Beta                                                                                            |
| -destination                                                                | generic/platform=iOS                                                                                  |
| -archivePath                                                                | /Users/Shared/Jenkins/Library/Developer/Xcode/Archives/2016-06-10/HCWeb 2016-06-10 17.29.28.xcarchive |
| archive                                                                     |                                                                                                       |
| | tee /Users/Shared/Jenkins/Library/Logs/gym/HCWeb-HCWeb\ Beta.log | xcpretty |                                                                                                       |
+-----------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------+
INFO [2016-06-10 17:29:28.20]: $ set -o pipefail && xcodebuild -workspace './HCWeb.xcworkspace' -scheme 'HCWeb Beta' -destination 'generic/platform=iOS' -archivePath '/Users/Shared/Jenkins/Library/Developer/Xcode/Archives/2016-06-10/HCWeb 2016-06-10 17.29.28.xcarchive' archive | tee /Users/Shared/Jenkins/Library/Logs/gym/HCWeb-HCWeb\ Beta.log | xcpretty
[SNIP]
INFO [2016-06-10 17:29:36.87]: ▸ Building HCWeb/HCWeb [AdHoc]
INFO [2016-06-10 17:29:36.87]: ▸ Check Dependencies
INFO [2016-06-10 17:29:36.88]: ▸ ❌  Code Sign error: No codesigning identities found: No codesigning identities (i.e. certificate and private key pairs) that match the provisioning profile specified in your build settings (“match AdHoc com.hotelscombined.findhotels.beta”) were found.
INFO [2016-06-10 17:29:36.88]: ▸ ** ARCHIVE FAILED **
INFO [2016-06-10 17:29:36.88]: ▸ The following build commands failed:
INFO [2016-06-10 17:29:36.88]: ▸ Check dependencies
INFO [2016-06-10 17:29:36.88]: ▸ (1 failure)
[SNIP]
▸ Building HCWeb/HCWeb [AdHoc]
▸ Check Dependencies

❌  Code Sign error: No codesigning identities found: No codesigning identities (i.e. certificate and private key pairs) that match the provisioning profile specified in your build settings (“match AdHoc com.hotelscombined.findhotels.beta”) were found.


** ARCHIVE FAILED **


The following build commands failed:
Check dependencies
(1 failure)
ERROR [2016-06-10 17:29:36.94]: Exit status: 65
INFO [2016-06-10 17:29:36.95]: -------------------
INFO [2016-06-10 17:29:36.95]: --- Step: slack ---
INFO [2016-06-10 17:29:36.95]: -------------------
INFO [2016-06-10 17:29:38.59]: Successfully sent Slack notification
WARN [2016-06-10 17:29:38.59]: Variable Dump:
INFO [2016-06-10 17:29:38.59]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta_HC", :SIGH_PROFILE_TYPE=>"ad-hoc"}
ERROR [2016-06-10 17:29:38.59]: Error building the application - see the log above
INFO [2016-06-10 17:29:38.60]: Successfully generated documentation to path '/Users/Shared/Jenkins/Home/jobs/HotelSearch_HotelsCombined_iPhone_beta/workspace/fastlane/README.md'

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | Switch to ios beta lane             | 0           |
| 4    | match                               | 5           |
| 5    | gym                                 | 14          |
| 6    | slack                               | 1           |
+------+-------------------------------------+-------------+

ERROR [2016-06-10 17:29:38.61]: fastlane finished with errors
/Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.46.2/lib/fastlane_core/ui/interface.rb:139:in `user_error!': [!] Error building the application - see the log above (FastlaneCore::Interface::FastlaneError)
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.46.2/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	from /Library/Ruby/Gems/2.0.0/gems/gym-1.6.3/lib/gym/error_handler.rb:53:in `handle_build_error'
	from /Library/Ruby/Gems/2.0.0/gems/gym-1.6.3/lib/gym/runner.rb:101:in `block in build_app'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.46.2/lib/fastlane_core/command_executor.rb:88:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.46.2/lib/fastlane_core/command_executor.rb:88:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/gym-1.6.3/lib/gym/runner.rb:97:in `build_app'
	from /Library/Ruby/Gems/2.0.0/gems/gym-1.6.3/lib/gym/runner.rb:12:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/gym-1.6.3/lib/gym/manager.rb:10:in `work'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/actions/gym.rb:22:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:197:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:182:in `block in execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:181:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:181:in `execute_action'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:107:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/fast_file.rb:140:in `method_missing'
	from Fastfile:35:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:160:in `try_switch_to_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:115:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/fast_file.rb:140:in `method_missing'
	from Fastfile:9:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:50:in `block in execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:46:in `chdir'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/runner.rb:46:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/commands_generator.rb:48:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane_core-0.46.2/lib/fastlane_core/ui/fastlane_runner.rb:26:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/commands_generator.rb:236:in `run'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/commands_generator.rb:18:in `start'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/lib/fastlane/cli_tools_distributor.rb:47:in `take_off'
	from /Library/Ruby/Gems/2.0.0/gems/fastlane-1.94.1/bin/fastlane:5:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

### Configuration Files

Fastfile:
```ruby
fastlane_version "1.91.0"

default_platform :ios

platform :ios do
  lane :beta_HC do |options|
    beta(scheme: "HCWeb Beta", api_token: "[REDACTED]", ipa: "HCWeb.ipa", create_profile: options[:create_profile])
  end

  lane :beta do |options|
    match(type: "adhoc", readonly: !(options[:create_profile]))
    if(!(options[:create_profile]))
      gym(scheme: options[:scheme], use_legacy_build_api: true)
      hockey(api_token: options[:api_token], ipa: options[:ipa])
    end
  end
end
```

Appfile:
```ruby
apple_id "[REDACTED]"

for_platform :ios do
  team_id [REDACTED]
  for_lane :beta_HC do
    app_identifier "com.hotelscombined.findhotels.beta"
  end
end
```

### Environment

fastlane version: fastlane 1.94.1

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? no

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? no

**Comment by TKBurner** [comment]
@evilDave Hoep you are doing well! Can you let me know if you unlocked your keychain before you ran this. That is likely the problem on Jenkins. Thanks!

**Comment by evilDave** [comment]
That's kind of what i thought, but the keychain is unlocked.
It does ask for access again when it runs via the command line as the jenkins user. And as I said, even running it via the command line and entering the password for the keychain if it asks for it there is still nothing stored in the keychain like there is in the login keychain when i run it as a regular user.

**Comment by TKBurner** [comment]
Hi @evilDave Thanks for following up. What happens if you set the `force` parameter of `match` to `true`?

Thanks!

**Comment by evilDave** [comment]
When I do that it fails with:
```
bash-3.2$ match adhoc -a com.hotelscombined.findhotels.beta --force

+-----------------------+---------------------------------------------------------+
|                             Summary for match 0.6.1                             |
+-----------------------+---------------------------------------------------------+
| app_identifier        | com.hotelscombined.findhotels.beta                      |
| force                 | true                                                    |
| git_url               | git@github.com:evilDave/HotelsCombined-Certificates.git |
| type                  | adhoc                                                   |
| git_branch            | master                                                  |
| username              | david.clark@hotelscombined.com                          |
| keychain_name         | login.keychain                                          |
| readonly              | false                                                   |
| verbose               | false                                                   |
| shallow_clone         | false                                                   |
| force_for_new_devices | false                                                   |
| skip_docs             | false                                                   |
+-----------------------+---------------------------------------------------------+

[13:41:09]: Cloning remote git repo...
[13:41:14]: 🔓  Successfully decrypted certificates repo
[13:41:14]: Verifying that the certificate and profile are still valid on the Dev Portal...
[13:41:18]: Installing certificate...

+-------------------------------------+------------------------------------------------+
|                                Summary for sigh 1.8.0                                |
+-------------------------------------+------------------------------------------------+
| app_identifier                      | com.hotelscombined.findhotels.beta             |
| adhoc                               | true                                           |
| development                         | false                                          |
| username                            | david.clark@hotelscombined.com                 |
| force                               | true                                           |
| cert_id                             | 2W3WN5GQB7                                     |
| provisioning_name                   | match AdHoc com.hotelscombined.findhotels.beta |
| ignore_profiles_with_different_name | true                                           |
| skip_install                        | false                                          |
| skip_fetch_profiles                 | false                                          |
| skip_certificate_verification       | false                                          |
+-------------------------------------+------------------------------------------------+

[13:41:19]: Starting login with user 'david.clark@hotelscombined.com'
[13:41:22]: Successfully logged in
[13:41:22]: Fetching profiles...
[13:41:25]: Certificate for Provisioning Profile 'match AdHoc com.hotelscombined.findhotels.beta' not available locally: 2W3WN5GQB7
[13:41:25]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[13:41:28]: No certificates for filter: Certificate ID: '2W3WN5GQB7'
[13:41:28]: 🔒  Successfully encrypted certificates repo

[!] Could not find a matching code signing identity for type 'AdHoc'. You can use cert to generate one:
https://github.com/fastlane/fastlane/tree/master/cert
```
I don't believe that I want/need to run cert again... I've already generated the signing identity as my regular user, indeed, my regular user can run the same command with no issues, works just fine. If I run cert again that would generate a new one no? Isn't this mess of creating certs and profiles exactly what the match command is supposed to handle? The certs and profiles are stored in the git repo and they should just be gotten out and installed.

Indeed, running without force ```match adhoc -a com.hotelscombined.findhotels.beta``` completes with success... but gym then fails with that error.

Frustrating.

Running the match command again with --verbose I can see the security commands to add the certs:
```
security import /var/folders/sl/rb8wdyyx2cn6_fpxz_vqd_j800007c/T/d20160621-40195-1m2rd32/certs/distribution/2W3WN5GQB7.cer -k ~/Library/Keychains/login.keychain -T /usr/bin/codesign -T /usr/bin/security&> /dev/null
```
does not match up with the command to find the developer relations cert:
```
security find-certificate -c 'Apple Worldwide Developer Relations Certification Authority' /Users/Shared/Jenkins/jenkins.keychain
```
indeed ~/Library/Keychains/login.keychain does not exist for this user. It would seem to me that it needs to go into the /Users/Shared/Jenkins/jenkins.keychain keychain.
Regardless, changing to:
```
security import /var/folders/sl/rb8wdyyx2cn6_fpxz_vqd_j800007c/T/d20160621-40195-1m2rd32/certs/distribution/2W3WN5GQB7.cer -k /Users/Shared/Jenkins/jenkins.keychain -T /usr/bin/codesign -T /usr/bin/security
```
Does not solve the problem:
```
security: SecKeychainItemImport: Unknown format in import.
```
If I login as my regular user, delete the distribution certificate from the keychain and run match I get an error pop up:
```
ERROR [2016-06-21 14:14:10.24]: There are no local code signing identities found. You can run `security find-identity -v -p codesigning` to get this output. This Stack Overflow thread has more information: http://stackoverflow.com/q/35390072/774. (Check in Keychain Access for an expired WWDR certificate: http://stackoverflow.com/a/35409835/774 has more info.
```
but it does continue on and install the certs from the repo. However If I run the commands which I assume do this installation:
```
security import /var/folders/j2/tr7b09fd5jd155t4zdnd05w40000gn/T/d20160621-41095-5lcisn/certs/distribution/2W3WN5GQB7.cer -k ~/Library/Keychains/login.keychain -T /usr/bin/codesign -T /usr/bin/security&> /dev/null
```
manually I get the same error as with the jenkins user.

**Comment by TKBurner** [comment]
Hi @evilDave Have you run `match developement` yet for this project? You need to do that first. Thanks!

**Comment by evilDave** [comment]
Yes I had done that. Not sure what difference that would make, the CI build should be using the distribution certificates not the development one.

I got fed up with this and `match nuke`ed it (probably not necessary), deleted the jenkins keychain and logged in as jenkins to let it create a login keystore in the normal place `~/Library/Keystores/login.keychain`.

I re-ran match on all my targets as my dev user to create and store the certs/profiles and then tried again running the lanes as jenkins. After a few teething problems (no interactive access to private key, keychain not remaining unlocked during all of build) it's finally sorted!

I really think that the issue is that somehow it's confused about which keychain to put the certs and keys in. Also I notice that no matter what, it always calls `security import` in the p12 file, *regardless* of whether or not the private key exists already. I.e. this runs every time:
```
security import /var/[blah]/GH2ZHKD3DJ.p12 -k ~/Library/Keychains/login.keychain -T /usr/bin/codesign -T /usr/bin/security&> /dev/null
```
Fortunately it seems to not reset the "Allow all applications to access this item" setting on the private key - this should really be calling `security` with the `-A` option to set that access when it gets added (I had to open the keychain and add permission to get past the "User interaction is not allowed" error).

Anyway, this has turned into an enormously verbose issue. I think it's solved, there were lots of little gotchas on the way that I probably should have documented better but I just needed it working.

TLDR; Deleted custom keychain and used jenkins login keychain.

Thanks @TKBurner for sticking with me and trying to help :)
